src/pipecat/pipeline/service_switcher.py

```diff
@@ -108,6 +90,43 @@ def __init__(self, services: List[FrameProcessor], strategy_type: Type[StrategyT
         self.services = services
         self.strategy = strategy
 
+    class ServiceSwitcherFilter(FunctionFilter):
+        """An internal filter that allows frames to pass through to the wrapped service only if it's the active service."""
+
+        def __init__(
+            self,
+            wrapped_service: FrameProcessor,
+            active_service: FrameProcessor,
+            direction: FrameDirection,
+        ):
+            """Initialize the service switcher filter with a strategy and direction."""
+
+            async def filter(_: Frame) -> bool:
+                return self.wrapped_service == self.active_service
+
+            super().__init__(filter, direction)
+            self.wrapped_service = wrapped_service
+            self.active_service = active_service
```

Comment:
nit: We should probably make these variables private: `_wrapped_service` and `_active_service`.

Reply:
Good call. Updating.